code/modules/client/client_procs.dm

```diff
@@ -104,6 +104,20 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 	if (href_list["player_ticket_panel"])
 		view_latest_ticket()
 		return
+	// Admin message
+	if(href_list["messageread"])
+		if(!isnum(href_list["messageread"]))
+			return
+		var/rounded_message_id = round(href_list["messageread"], 1)
+		var/datum/db_query/query_message_read = SSdbcore.NewQuery(
+			"UPDATE [format_table_name("messages")] SET type = 'message sent' WHERE targetckey = :player_key AND id = :id",
+			list("id" = rounded_message_id, "player_key" = usr.ckey)
+		)
+		if(!query_message_read.warn_execute())
+			qdel(query_message_read)
+			return
+		qdel(query_message_read)
```

Comment:
what's the point of the early return here? i think in the old code that you moved it from, there was probably a reason to prematurely terminate the entire chain if `warn_execute()` failed... but i think this needs to be retooled here to either _always_ return or _always_ continue the rest of this proc chain agnostic of `warn_execute()` since it seems pretty silly in the current state (as in no different behavior is accomplished even if we are listening for the outcome of that proc).

Reply:
true, we should always return here, good catch